### ISSUE TITLE
no_std Version

### ISSUE THREAD

**Issue by pinkforest** [khonsulabs/justjson#1]
Would be nice if you're up to it :)

+ alloc | heapless

For both embedded and wasm targets

**Comment by ecton** [comment]
I hadn't heard of heapless before. What advantages does heapless have over just using alloc, assuming I only need Vec/String?

I'm definitely up for this -- I only have one no_std crate so far, so I'm still new to that ecosystem.

**Comment by ecton** [comment]
Oh, I see (still waking up). I just realized the major difference. I think I can only adopt heapless for the Document type. Because of the nesting nature of Value, I don't think I can use Heapless without drastic changes to how the Value type works.

**Comment by ecton** [comment]
`no_std` using alloc is now supported after 346548569b214138d4177d0e00960a752de27082.

I'm keeping this open for the heapless idea for Document.

**Comment by pinkforest** [comment]
Seems to work

```
$ rustup default stable
$ rustup target add thumbv7em-none-eabi
$ cargo +stable build --target thumbv7em-none-eabi --lib --release --no-default-features
    Finished release [optimized + debuginfo] target(s) in 0.03s
```

btw thumbv7em-none-eabi doesn' thave std so it's nice to test with :)

```yaml
  build-nostd:
    name: Build on no_std target (thumbv7em-none-eabi)
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: dtolnay/rust-toolchain@master
        with:
          toolchain: stable
          targets: thumbv7em-none-eabi
      - run: cargo build --target thumbv7em-none-eabi --lib --release --no-default-features
      - run: cargo build --target thumbv7em-none-eabi --lib --release
```

p.p.s. I would put the `extern alloc` crate behind feature cfg `alloc`

**Comment by ecton** [comment]
The main branch now [has heapless support](https://github.com/khonsulabs/justjson/blob/968b7860d7263a283f87f104d14109cff33c371a/examples/heapless.rs) through the document parsing interface.

While it's not as easy as interacting with the Value type, it at least works :) I haven't come up with a decent approach to implementing the value type, due its nesting nature.

**Comment by ecton** [comment]
I've published [v0.2.0](https://github.com/khonsulabs/justjson/releases/tag/v0.2.0) which supports all variations of std, alloc, and heapless flags, including no features being enabled.

Thank you for the suggestion and snippets along the way!
